frontend/src/services/eventHandlers/hubHandler.js

```diff
@@ -1,12 +1,63 @@
+import { EventBus } from "@/eventBus";
+import { Event } from "@/events";
 import Player from "@/modules/common/entities/player";
 import { store } from "@/store";
 
 // Event handler for Hub API
 export default class HubHandler {
     handle(payload) {
         if (payload.connectedClients != null) {
-            const players = payload.connectedClients.map(client => new Player(client.clientUUID.UUID, client.nickname, client.joinOrder));
+            const players = payload.connectedClients.map(
+                (client) =>
+                    new Player(
+                        client.clientUUID.UUID,
+                        client.nickname,
+                        client.joinOrder
+                    )
+            );
+
+            this.handlePlayerJoined(players);
+            this.handlePlayerLeft(players);
+
             store.commit("application/setPlayers", players);
         }
     }
+
+    handlePlayerJoined(players) {
+        const currentPlayers = store.getters["application/getPlayers"];
+        if (currentPlayers.length !== 0) {
+            const clientUuids = players.map((player) => player.uuid);
+            const currentUuids = currentPlayers.map((player) => player.uuid);
+
+            let playerJoinGame = clientUuids.filter(
```

Comment:
Can we use the functional programming way of doing this because we are already using filter. ie:
clientUuids.filter(uuid => !currentUuids.include(uuid)).map(uuid => {
players.filter(p => p.uuid === uuid))
}.foreach(player => {
EventBus.$emit(Event.PLAYER_JOIN, player);
})